bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2ApiRpc.java

```diff
@@ -170,7 +170,6 @@ public ShellyDeviceProfile getDeviceProfile(String thingType, @Nullable ShellySe
         }
 
         Shelly2GetConfigResult dc = apiRequest(SHELLYRPC_METHOD_GETCONFIG, null, Shelly2GetConfigResult.class);
-        profile.isGen2 = true;
```

Comment:
There is not need to overwrite the flag here, or in fact it's wrong.
The flag is initialized in ShellyDeviceProfile() based on thing type and the GEN2 API is already bound, otherwise we could never reach this place here.

Reply:
Sorry, but I still don't fully get it. In line 163 in this method:
```java
ShellyDeviceProfile profile = thing != null ? getProfile() : new ShellyDeviceProfile();
```

So if `thing` is null, `profile` will be uninitialized after being created with the empty constructor. Won't this become a problem then, at least I don't see a path leading to `isGen2` being set?

Reply:
This was resolved without a comment, can you help me out here? 🙂 Also, the same change is made in #17163, so I think that PR needs to be merged first, as otherwise it may cause problems when cherry-picking it in `4.2.x`.

Reply:
This code is part of the Gen2 handler, so if we reach that line here the APIIv2  has already been created and isGen2 was initialised by the initFromThingType(). So no need to do it again here and in fact that would be the wrong position.